src/ssl.c

```diff
@@ -21121,7 +21121,8 @@ void wolfSSL_AES_cbc_encrypt(const unsigned char *in, unsigned char* out,
 
     WOLFSSL_ENTER("wolfSSL_AES_cbc_encrypt");
 
-    if (key == NULL || in == NULL || out == NULL || iv == NULL) {
+    if (key == NULL || in == NULL || out == NULL || iv == NULL ||
+        (int) len <= 0 ) {
```

Comment:
Casting a word32 to int to allow a <0 check seems odd. Recommend changing this to just `|| len == 0`. You may also need to change your `lenN = -1;` tests below... The change also restricts this function to max int32.

Reply:
Excellent notes! Will push update shortly.